src/libsync/owncloudpropagator.cpp

```diff
@@ -386,33 +390,29 @@ void OwncloudPropagator::start(const SyncFileItemVector& items)
                 }
             } else {
                 PropagateDirectory* currentDirJob = directories.top().second;
-                currentDirJob->append(dir);
+                currentDirJob->appendJob(dir);
             }
             directories.push(qMakePair(item->destination() + "/" , dir));
-        } else if (PropagateItemJob* current = createJob(item)) {
+        } else {
             if (item->_instruction == CSYNC_INSTRUCTION_TYPE_CHANGE) {
                 // will delete directories, so defer execution
-                directoriesToRemove.prepend(current);
+                directoriesToRemove.prepend(createJob(item));
```

Comment:
directoriesToRemove could actually be a vector of items (no need to create the job before)
(that's just a further optimisation.)